src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -0,0 +1,173 @@
+import React, { useEffect, useState, useCallback, useMemo } from "react";
+import { Form, Modal, Button } from "react-bootstrap";
+import { useForm } from "react-hook-form";
+import ImageInput from "components/atoms/ImageInput";
+
+import { RoomTemplate, ROOM_TEMPLATES } from "settings";
+import { RoomEditModalProps } from "./RoomEditModal.types";
+
+import "./RoomEditModal.scss";
+
+export const RoomEditModal: React.FC<RoomEditModalProps> = ({
+  isVisible,
+  onClickOutsideHandler,
+  room,
+  submitHandler,
+  deleteHandler,
+}) => {
+  const [selectedTemplate, setSelectedTemplate] = useState<string>(
+    room.template!
+  );
+
+  const [roomTemplate, setRoomTemplate] = useState<RoomTemplate | null>(null);
+
+  useEffect(() => {
+    const template = ROOM_TEMPLATES.find(
+      (i) => i.template === selectedTemplate
+    );
+    setRoomTemplate(template!);
+  }, [selectedTemplate]);
+
+  const {
+    register,
+    watch,
+    handleSubmit,
+    setValue,
+    formState: { isSubmitting, dirty },
+  } = useForm({
+    defaultValues: {
+      title: room.title,
+      url: room.url,
+      description: room.description,
+      template: room.template,
+    },
+  });
+
+  const values = watch();
+
+  const handleImageChange = useCallback(
+    (val: string) => {
+      setValue("image_url", val, false);
+    },
+    [setValue]
+  );
+
+  function onSubmit() {
+    return submitHandler(values, room.roomIndex!);
+  }
```

Comment:
Change this to a `const onSubmit = () =>` format to match our style elsewhere + `useCallback`